### ISSUE TITLE
Not up-to-date focus in high resolution when changing focus in some windows

### ISSUE THREAD

**Issue by vincent-grosbois** [ihhub/fheroes2#2598]
You can see this when you do the following:
- put resolution in 1024*768 or above
- go to castle screen
- hire hero
- --> new Hero doesn't appear in the hero bar

Something similar in hero window:
- select hero
- change hero with arrows
- hero is not on focus in the hero bar
- (focus doesn't come back to the hero when you exit)

I think there must be other bugs, where basically the righ-side bar is never updated when actions are performed in a sub-window.
Note that this can be fixed, like I've done for kingdom view: 
https://github.com/ihhub/fheroes2/pull/2597
Basically by redrawing the world map, then redrawing the window

**Comment by Branikolog** [comment]
Hi, @vincent-grosbois.
Cannot say, it's a bug...
As original game doesn't support higher resolutions, the ability to watch GUI and World map surrounding windows mentioned above is the case appeared in fheroes2. So in fheroes2 it was implemented, that all the GUI of the game is not updated, while you perform any actions in the window. 
Anyway, the idea is good and I hope game engine allows us to implement it.
Also, I want to add to your suggestion, we should implement interaction with GUI, so, if you click on castle icon in a bar to the right, we can switch castles in current window without pressing arrow buttons or closing it.

**Comment by Branikolog** [comment]
Between, changing focus on exit the window, while switching castles or heroes with arrow buttons, while hero/castle window is open should be a separate issue, as we can easily fix it, i hope.
